### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by yui-knk**

### CODE REVIEW

**Comment by yui-knk** [comment]
All tests (except ruby-head) backs to be green.
Tests on ruby-head never to be green, because `Ruby 2.4.0 isn't supported by this pry-doc version`.
I feel it is better to allow fail ruby-head 😸

**Comment by kyrylo** [comment]
We should `allow_failures` on Travis, then.

**Comment by yui-knk** [comment]
🙆 ❓

File: lib/pry-doc.rb [hunk 1]
```diff
@@ -56,15 +56,13 @@ def self.receiver_notation_for(meth)
     # @return [Array] The aliases of a method if it exists
     #                 otherwise, return empty array
     def self.aliases(meth)
-      host        = is_singleton?(meth) ? meth.receiver : meth.owner
-      method_type = is_singleton?(meth) ? :method : :instance_method
```
Comment (kyrylo) [inline lib/pry-doc.rb]: Perhaps I'm missing something, but why do we not call `is_singleton?` anymore?

Comment (yui-knk) [inline lib/pry-doc.rb]: `owner` of class method is singleton class. So we can call `#method ` on either case (instance methods or class methods)

```ruby
class Test
  def a
  end

  def b
  end

  class << self
    def c
    end

    def d
    end
  end
end

a_method = Test.instance_method(:a)
c_method = Test.method(:c)

a_method = Test.instance_method(:a)
# => #<UnboundMethod: Test#a>
c_method = Test.method(:c)
# => #<Method: Test.c>
a_method.owner
# => Test
c_method.owner
# => #<Class:Test>
a_method.owner.instance_method(:a)
# => #<UnboundMethod: Test#a>
c_method.owner.instance_method(:c)
# => #<UnboundMethod: #<Class:Test>#c>
```

**Comment by kyrylo** [comment]
👯

File: lib/pry-doc.rb [hunk 2]
```diff
@@ -56,15 +56,13 @@ def self.receiver_notation_for(meth)
...
+      host        = meth.owner
+      method_type = :instance_method
+      methods = (host.instance_methods + host.private_instance_methods).uniq
```
Comment (banister) [inline lib/pry-doc.rb]: do we also need to add `host.protected_instance_methods` into here as well?

Comment (yui-knk) [inline lib/pry-doc.rb]: Not needed

```ruby
class A
  def pub
  end

  def pro
  end

  protected :pro

  def pri
  end

  private :pri
end

A.instance_methods(false)
=> [:pub, :pro]
A.private_instance_methods(false)
=> [:pri]
```

**Pull request merged**